### PR TITLE
[DPE-8571] | Scale HA integration test

### PR DESCRIPTION
**Pull request by BON4**
### Title

Add `scale-down, scale up, check that member returned with data` integration test.

### Summary

This PR adds a helper function `scale_sequentially_to()` and updates scaling-related integration tests to improve reliability and consistency during cluster resize operations.

### CODE REVIEW

**Review by Zvirovyi — COMMENTED** [review]
question There's charm error detected in test_scaling:
```
+ .apps['cassandra'].units['cassandra/1'].workload_status.current = 'error'
+ .apps['cassandra'].units['cassandra/1'].workload_status.message = 'hook failed: "storage-detaching"'
+ .apps['cassandra'].units['cassandra/1'].juju_status.current = 'idle'
---------------------------- Captured log teardown -----------------------------
INFO     jubilant:_juju.py:237 cli: juju debug-log --model test-cas --limit 300
=========================== short test summary info ============================
FAILED tests/integration/test_scaling.py::test_single_node_scale_down - TimeoutError: wait timed out after 600s
```

Can you check this out, @BON4? Thanks!
If it's a bug, we can address it in the different ticket. I just don't want to merge error-throwing integration test.

File: tests/integration/helpers.py [hunk 1]
```diff
@@ -378,3 +378,54 @@ def assert_rows(wrote: dict[int, str], got: dict[int, str]) -> None:
     """Assert rows are equal."""
     assert len(got) == len(wrote), f"Expected {len(wrote)} rows, got {len(got)}"
     assert got == wrote, "Row data mismatch"
+
+
+def scale_sequentially_to(juju: jubilant.Juju, app_name: str, n: int) -> None:
```
Comment (deusebio) [inline tests/integration/helpers.py:553]: **praise** really nice helper to scale up and down of arbitrary units! Well done! I would just also add the batching possibility when scaling up, I mean, we don't support multiple units scale down, but we do support this on scaling up (e.g. add 2 units). 

If we only tests the one-by-one, I'm afraid we may miss some edge cases

File: tests/integration/test_scaling.py [hunk 2]
```diff
@@ -31,8 +37,7 @@ def test_scale_up(juju: jubilant.Juju, app_name: str) -> None:
 
     old_units = set(juju.status().apps[app_name].units.keys())
 
-    juju.add_unit(app_name, num_units=2)
-    juju.wait(jubilant.all_active, timeout=1200)
+    scale_sequentially_to(juju, app_name, 3)
```
Comment (deusebio) [inline tests/integration/test_scaling.py:40]: **todo** I would still like to test a batch add of units (of 2)

**Review by deusebio — COMMENTED** [review]
Looks good! I would just like to still keep the possibility of adding units in batch. 

Thanks!

**Comment by BON4** [comment]
It seems there is the bug in data_platform libs that occurs only for juju 3.5. I've added temporary fix. And created an issue report https://github.com/canonical/data-platform-libs/issues/243#issue-3527889114

File: tests/integration/helpers.py (hunk 1 above)
Comment (BON4) [reply tests/integration/helpers.py:553]: Fixed. Now function scales up all at once.

**Review by deusebio — APPROVED** [review]
LGTM! Great job!

**Review by Zvirovyi — APPROVED** [review]
Just fix the merge conflict.
